pkg/registry/pod/etcd/etcd.go

```diff
@@ -218,6 +220,38 @@ type LogREST struct {
 	kubeletConn client.ConnectionInfoGetter
 }
 
+// Max length read from the body of a log location which returns error status
+const (
+	maxReadLength = 1000
+)
+
+// A log location response checker to transform the error.
+type LogLocationResponseChecker struct {
+	podName string
```

Comment:
It's coupled to the backend that returns pod logs today, the Kubelet, which
isn't as generic as moving it to generic/rest would imply.  The kubelet API
isn't even really designed.  So if the kubelet changes, and someone goes
and changes generic/rest, that would be less correct than this still being
slightly coupled to pods.

On Wed, Sep 23, 2015 at 5:48 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/registry/pod/etcd/etcd.go
> <https://github.com/kubernetes/kubernetes/pull/13722#discussion_r40262295>
> :
>
> > @@ -218,6 +220,38 @@ type LogREST struct {
> >  	kubeletConn client.ConnectionInfoGetter
> >  }
> >
> > +// Max length read from the body of a log location which returns error status
> > +const (
> > +	maxReadLength = 1000
> > +)
> > +
> > +// A log location response checker to transform the error.
> > +type LogLocationResponseChecker struct {
> > +	podName string
>
> This isn't really specific to pods, actually, this is only used for the
> error message, right?
>
> Why not move this object into registry/generic/rest and then you don't
> have to make the fake, either?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13722/files#r40262295>.
>